pkg/scheduler/internal/cache/cache.go

```diff
@@ -506,17 +506,19 @@ func (cache *cacheImpl) AddPod(pod *v1.Pod) error {
 	currState, ok := cache.podStates[key]
 	switch {
 	case ok && cache.assumedPods.Has(key):
+		// When assuming, we've already added the Pod to cache,
+		// Just update here to make sure the Pod's status is up-to-date.
+		if err = cache.updatePod(currState.pod, pod); err != nil {
+			klog.ErrorS(err, "Error occurred while updating pod")
+		}
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
 			// The pod was added to a different node than it was assumed to.
 			klog.InfoS("Pod was added to a different node than it was assumed", "podKey", key, "pod", klog.KObj(pod), "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
-			if err = cache.updatePod(currState.pod, pod); err != nil {
-				klog.ErrorS(err, "Error occurred while updating pod")
-			}
-		} else {
-			delete(cache.assumedPods, key)
-			cache.podStates[key].deadline = nil
-			cache.podStates[key].pod = pod
+			return nil
 		}
+		delete(cache.assumedPods, key)
```

Comment:
Why we still need this, we have removed and readded the pod.

Reply:
Where do we readd the Pod?

Reply:
When deleting, https://github.com/kubernetes/kubernetes/blob/0f37b3120643580f632ca12d3e174e7ec447948c/pkg/scheduler/internal/cache/cache.go#L492-L493

When adding, https://github.com/kubernetes/kubernetes/blob/0f37b3120643580f632ca12d3e174e7ec447948c/pkg/scheduler/internal/cache/cache.go#L449-L452

Reply:
Now, understood what you mean. Removed it.